src/hotspot/share/utilities/lineReader.hpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_LINEREADER_HPP
+#define SHARE_UTILITIES_LINEREADER_HPP
+
+#include "memory/allocation.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+// This class is a wrapper around fgets() for reading *reasonably long*
+// text lines (up to LineReader::MAX_LEN-1 chars) from a FILE*.
+//
+// MAX_LEN is currently 4M. This should be enough for any practical use
+// of text-based input files for HotSpot. Don't use LineReader if it's
+// possible for valid lines to be longer than this limit.
+class LineReader : public StackObj {
```

Comment:
Suggestion: LineReader should also track the line count.

Reply:
Done. As a result, I removed `ClassListParser::_line_no`.